src/V3EmitC.cpp

```diff
@@ -3376,29 +3377,25 @@ class EmitCTrace : EmitCStmts {
         string full = ((m_funcp->funcType() == AstCFuncType::TRACE_FULL
                         || m_funcp->funcType() == AstCFuncType::TRACE_FULL_SUB)
                        ? "full":"chg");
-        bool emitWidth = false;
+        bool isArray = false;
         if (nodep->dtypep()->basicp()->isDouble()) {
             puts("vcdp->"+full+"Double");
         } else if (nodep->isWide() || emitTraceIsScBv(nodep) || emitTraceIsScBigUint(nodep)) {
-            puts("vcdp->"+full+"Array");
-            emitWidth = true;
+            isArray = true;
+            puts("vcdp->" + full + "Array<" + cvtToStr(nodep->declp()->widthMin() % 32) + ">");
         } else if (nodep->isQuad()) {
-            puts("vcdp->"+full+"Quad");
-            emitWidth = true;
-        } else if (nodep->declp()->bitRange().ranged()
-                   // 1 element smaller to use Bit dump
-                   && nodep->declp()->bitRange().elements() != 1) {
-            puts("vcdp->"+full+"Bus");
-            emitWidth = true;
+            puts("vcdp->"+full+"Quad<"+cvtToStr(nodep->declp()->widthMin())+ ">");
+        } else if (nodep->declp()->widthMin() > 1) {
+            puts("vcdp->"+full+"Bus<"+cvtToStr(nodep->declp()->widthMin())+ ">");
```

Comment:
clang-format this block
```suggestion
            puts("vcdp->" + full + "Quad<" + cvtToStr(nodep->declp()->widthMin()) + ">");
        } else if (nodep->declp()->widthMin() > 1) {
            puts("vcdp->" + full + "Bus<" + cvtToStr(nodep->declp()->widthMin()) + ">");
```